File: src/zork/models/entities/Entity.java
package zork.models.entities;

import java.util.EnumMap;
import java.util.Map;

/**
 *
 * @author d.peters
 */
public class Entity {

    /**
     *
     */
    private String name;
    
    /**
     * 
     */
    protected Map<EStats,Stat> stats;

    /**
     *
     */
    private int age;

    /**
     *
     * @param name
     * @param age
     */
    public Entity(String name, int age) {

        this.name = name;
        this.age = age;
        this.stats = new EnumMap<>(EStats.class);

    }

    /**
     *
     * @return
     */
    public String getName() {
        return name;
    }

    /**
     *
     * @return
     */
    public int getAge() {
        return age;
    }

    /**
     * 
     * @return 
     */
    public Map<EStats, Stat> getStats() {
        return stats;
    }
    

}

File: src/main/kotlin/ch/peters/daniel/zork/models/entities/base/Entity.kt
package ch.peters.daniel.zork.models.entities.base

import ch.peters.daniel.zork.models.entities.Moveable

/**
 * Basic Entity.
 *
 * @author Daniel Peters <daniel.peters.ch@gmail.com>
 * @version 1.1.0
 */
abstract class Entity(var name: String, var age: Int, attack: Stat) : Moveable {
  var alive: Boolean
  var stats: MutableMap<StatType, Stat>

  fun die() {
    alive = false
    System.out.println("$name died.")
  }

  /**
   * Resurrect entity.
   */
  fun resurrect() {
    if (!alive) {
      alive = true
      System.out.println("$name resurrected.")
    } else {
      System.out.println("You can't resurrect an alive person you dummy!")
    }
  }

  init {
    alive = true
    stats = HashMap()
    stats[StatType.ATTACK] = attack
  }
}
